app/backend/src/core/controllers/appraiser/BlockAppraiser.test.ts

```diff
@@ -0,0 +1,189 @@
+    const block = await getBlockWithTx(
+      595593,
+      "0x6f69459c4196222a227d4e4c79787f4ffcb51b47f7f545fbc01a0753175aa7b",
+    );
+
+    const value = await blockAppraiser.value(block!);
+
+    expect(value).toEqual({
+      transferSummary: {
+        mapped: [
+          // Fee:
+          {
+            contractAddress: assets["STRK"].address!,
+            usdTotalValue: 0.359736409417677808 * fakePrice.priceUsd,
+          },
+        ],
+        unmapped: [],
+      },
+      blockRewardSummary: noBlockRewards,
+    });
+  });
+
+  // https://voyager.online/tx/0x10eebefaef1771ceac9a7ace258e5efeeb5ef27bfa12559c3884173ac0b28ac
+  it("handles simple transfer", async () => {
+    const block = await getBlockWithTx(
+      595564,
+      "0x10eebefaef1771ceac9a7ace258e5efeeb5ef27bfa12559c3884173ac0b28ac",
+    );
+
+    const value = await blockAppraiser.value(block!);
+
+    expect(value).toEqual({
+      transferSummary: {
+        mapped: [
+          // Transfer:
+          {
+            contractAddress: assets["STRK"].address!,
+            usdTotalValue: 470.72 * fakePrice.priceUsd,
+          },
+          // Fee:
+          {
+            contractAddress: assets["ETH"].address!,
+            usdTotalValue: 0.00025715858141008 * fakePrice.priceUsd,
+          },
+        ],
+        unmapped: [],
+      },
+      blockRewardSummary: noBlockRewards,
+    });
+  });
+
+  // https://voyager.online/tx/0x79c06c1550f98797c0ea76228f2e54d4b5b495f85ac85e555e4a1b111453e04
+  it("handles swap", async () => {
+    const block = await getBlockWithTx(
+      595593,
+      "0x79c06c1550f98797c0ea76228f2e54d4b5b495f85ac85e555e4a1b111453e04",
+    );
+
+    const value = await blockAppraiser.value(block!);
+
+    expect(value).toEqual({
+      transferSummary: {
+        mapped: [
+          // Max of transfers:
+          {
+            contractAddress: assets["STRK"].address!,
+            usdTotalValue: 20.0 * fakePrice.priceUsd,
+          },
+          // Fee: TODO: enable once swap fees are properly handled by StarknetClient
+          // {
+          //   contractAddress: assets["STRK"].address!,
+          //   usdTotalValue: 0.978754939960334552 * fakePrice.priceUsd,
+          // },
+        ],
+        unmapped: [],
+      },
+      blockRewardSummary: noBlockRewards,
+    });
+  });
+
+  // https://voyager.online/tx/0x4c213caef8e503a7b3c54edf18ebca1d6117e1821d68ea5e6db1ef63aee6f04
+  it("handles swap with unmapped asset", async () => {
+    const block = await getBlockWithTx(
+      613171,
+      "0x4c213caef8e503a7b3c54edf18ebca1d6117e1821d68ea5e6db1ef63aee6f04",
+    );
+
+    const value = await blockAppraiser.value(block!);
+
+    expect(value).toEqual({
+      transferSummary: {
+        mapped: [
+          // Max of transfers:
+          {
+            contractAddress: assets["DAI"].address!,
+            // TODO: improve precision, should be 68.156400247938_186314_
```

Comment:
Don't think it's critical, but just to point out a specific example of the issue.